### PR TITLE
jnp.eye: handle larger-than int32 offsets

### PR DESCRIPTION
**Pull request by jakevdp**
This fixes a corner case probed by the array API tests; see #16099 (Part of https://github.com/google/jax/issues/18353)

Also did the same for `jnp.tri`, but I left out the tests because numpy issues a warning for large integer offsets.

### CODE REVIEW

File: tests/lax_numpy_test.py [hunk 1]
```diff
@@ -2025,7 +2025,20 @@ def testTrapz(self, yshape, xshape, dtype, dx, axis):
     dtype=default_dtypes,
     n=[0, 4],
     m=[None, 0, 1, 3, 4],
-    k=list(range(-4, 4)),
+    k=[*range(-4, 4), -2**100, 2**100],
+  )
+  def testEye(self, n, m, k, dtype):
```
Comment (froystig) [inline tests/lax_numpy_test.py:2030]: Call for an optometrist?

**Review by froystig — APPROVED** [review]
